### ISSUE TITLE
GPU type and GPU nums and total training time on Transformer-XL, GPT-2

### ISSUE THREAD

**Issue by kenoharada** [sail-sg/Adan#32]
Hi! Thank you for sharing your code.

I would like to know for each Transformer-XL, GPT-2 settings.
- which GPU did you use?
- how many GPUs are used for training
- total training time

I saw logs, but I didn't figure out the exact number
https://github.com/sail-sg/Adan/tree/main/gpt2#results-and-logs-on-gpt2-345m
https://github.com/sail-sg/Adan/blob/main/gpt2/pretrain.sh
https://github.com/sail-sg/Adan/tree/main/NLP/Transformer-XL/exp_results

Thank you!

**Comment by XingyuXie** [comment]
@kenoharada 
Thanks for your interest.

We use 4 Nvidia V100 GPUs for Transformer-XL, and for the GPT-2 on code generalization, we use the 8 A100 GPUs during training. 

BZW, we still have some results on GPT-2 on that openwebtext data. The results are the following:

|                 | Val. Loss | Val. PPL | Steps | Model and Data Parallel |
| :-------------- | :-------: | :------: | :---: | :---------------------: |
| Adam (baseline) |   2.70    |   15.0   | 160k  |          false          |
| Adan            | **2.65**  | **14.3** | 160k  |          false          |
| 1-bit Adam      |   2.69    |   14.8   | 160k  |          false          |
| 1-bit Lamb      |   2.69    |   14.8   | 160k  |          false          |
| 0/1 Adam        |   2.69    |   14.8   | 160k  |          false          |

|                 | Val. Loss | Val. PPL | Steps | Model and Data Parallel |
| :-------------- | :-------: | :------: | :---: | :---------------------: |
| Adam (official) |   2.82    |   16.8   | 320k  |          true           |
| Adam (baseline) |   2.87    |   17.6   | 160k  |          true           |
| Adan            | **2.79**  | **16.3** | 160k  |          true           |

We run the experiments based on DeepSpeed https://www.deepspeed.ai/tutorials/megatron/
with 16 A100 GPUs. And we attach the configure file on the attachment.

Thanks again!

[gpt-adan.zip](https://github.com/sail-sg/Adan/files/11358542/gpt-adan.zip)

**Comment by kenoharada** [comment]
Thank you very much for sharing!

Your various experiments are so helpful to understand your optimizer's strong points.

Thank you!
